### PR TITLE
feat(alerts): Alert Wizard V3 template selector

### PR DESCRIPTION
**Pull request by taylangocmen**
This adds a dropdown in the metric selector in the alert wizard that can change the aggregate function, same way you would select an aggregate in the radio button selection step before the alert form step. Aggregate is changed in the url and the dropdown field follows the value from the url.

Wizard selector for aggregate:
Screen Shot 2022-04-06 at 1 51 14 PM

Expanded with all values:
Screen Shot 2022-04-06 at 1 50 24 PM

### CODE REVIEW

**Comment by evanpurkhiser** [comment]
@vuluongj20 is there not a newer type of dropdown that we can use here?

I though the categories / sub-categories thing looked different

File: static/app/views/alerts/incidentRules/wizardField.tsx [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+  router: InjectedRouter;
+  alertType?: AlertType;
+  /**
+   * Optionally set a width for each column of selector
+   */
+  columnWidth?: number;
+  inFieldLabels?: boolean;
+};
+
+const menuOptions = [
```
Comment (scttcper) [inline static/app/views/alerts/incidentRules/wizardField.tsx:38]: a type might be nice

File: static/app/views/alerts/incidentRules/wizardField.tsx [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+  inFieldLabels,
+  alertType,
+  ...fieldProps
+}: Props) {
+  const selected =
+    menuOptions.find(
+      op =>
+        AlertWizardRuleTemplates[op.key]?.aggregate === location.query.aggregate &&
+        AlertWizardRuleTemplates[op.key]?.dataset === location.query.dataset &&
+        AlertWizardRuleTemplates[op.key]?.dataset === location.query.dataset
```
Comment (scttcper) [inline static/app/views/alerts/incidentRules/wizardField.tsx:123]: can we destructure something here

File: static/app/views/alerts/incidentRules/wizardField.tsx [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+    css`
+      color: ${p.theme.textColor};
+      &:hover,
+      &:focus {
+        color: ${p.theme.textColor};
+      }
+    `}
+`;
+
+PresetButton.defaultProps = {
```
Comment (scttcper) [inline static/app/views/alerts/incidentRules/wizardField.tsx:258]: this is funky

**Review by scttcper — APPROVED** [review]
seems okay

**Pull request merged**